service/src/test/java/uk/nhs/adaptors/gp2gp/common/configuration/ObjectMapperConfigTest.java

```diff
@@ -0,0 +1,28 @@
+package uk.nhs.adaptors.gp2gp.common.configuration;
+
+import com.fasterxml.jackson.core.StreamReadConstraints;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ObjectMapperConfigTest {
+
+    @Test
+    void objectMapperConfigTest() {
+
+        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(ObjectMapperConfig.class);
+
+        ObjectMapper objectMapper = context.getBean(ObjectMapper.class);
+
+        StreamReadConstraints constraints = objectMapper.getFactory().streamReadConstraints();
+
+        assertEquals(Integer.MAX_VALUE, constraints.getMaxStringLength(),
+                     "Expected maxStringLength to be set to Integer.MAX_VALUE");
+
+        context.close();
+    }
```

Comment:
This test is wayyyy too low level in my view.

Somebody coming to this test should understand the context and the reasoning behind why this is valuable to the adaptor.

At the moment, they are told that `MaxStringLength` needs to be MAX_LENGTH. But no context as to why?!

Also, the test provides no _real_ proof that the original defect has been fixed, or that it won't regress. The test is coupled to an implementation detail as opposed to the behaviour we want our adaptor to have.

Reply:
this test was to address PiTest complain about the lack of tests for ObjectMapperConfig class objectMapper method. 
objectMapper method is itself very technical as it is a config bean. I can add a better description in assertEquals to give a better context... also, I will make the test name more descriptive which should give more context.

Reply:
I don't think you have read and fully understood what I wrote Ole. Adding better description won't fix my underlying issue.